avocado/utils/astring.py

```diff
@@ -219,7 +219,7 @@ def string_safe_encode(string):
     try:
         return string.encode("utf-8")
     except UnicodeDecodeError:
-        return string.decode("utf-8").encode("utf-8")
+        return string.decode("utf-8", "ignore").encode("utf-8")
```

Comment:
Actually you are right about the `Encode`. Anyway I'll add a selftest demonstrating all variants in v2 and yes, I think it should be included as the meaning of this function is to encode any kind of string into the most appropriate representation so allowing broken strings could be useful.